src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,369 @@
+      if (!inputs.isKickerIRBlocked) {
+        intakeState = IntakeState.EMPTY;
+        leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+        this.intakeGamePiece();
+      }
+    }
+  }
+
+  public Command getSystemCheckCommand() {
+    /*
+     * Order of what will be tested
+     * 1. make sure everything works at right velocities
+     * 2. then check that all the states it goes through when it gets a note are correct  
+     *
+     */
+    return Commands.sequence(
+      getVelocitiesCheckCommand(),
+      getStatesCheckCommand()
+    ).until(() -> !FaultReporter.getInstance().getFaults(SUBSYSTEM_NAME).isEmpty())
+    .andThen(Commands.runOnce(this::turnIntakeOff))
+    .withName(SUBSYSTEM_NAME + "SystemCheck");
+  }
+
+
+  private Command getVelocitiesCheckCommand() {
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> {
+          for (int i = 0; i < 4; i++) {
+            checkMotorVelocity(intakeMotors[i]);
+          }
+        })
+      )
+    );
+  }
+
+  private void checkMotorVelocity(IntakeMotor motor) {
+    if (motor == IntakeMotor.RIGHT_ROLLER) {
+      if (inputs.rightRollerVelocityRotationsPerSecond 
+      - inputs.rightRollerReferenceVelocityRPS 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Right Roller Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.LEFT_ROLLER) {
+      if (inputs.leftRollerVelocityRotationsPerSecond 
+      - inputs.leftRollerReferenceVelocityRPS 
+      > IntakeConstants.ROLLER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Left Roller Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.DRUM) {
+      if (inputs.drumVelocityRotationsPerSecond 
+      - inputs.drumReferenceVelocityRPS 
+      > IntakeConstants.DRUM_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Drum Too Slow"
+        );
+      }
+    } else if (motor == IntakeMotor.KICKER) {
+      if (inputs.kickerVelocityRotationsPerSecond 
+      - inputs.kickerReferenceVelocityRPS 
+      > IntakeConstants.KICKER_VELOCITY_TOLERANCE) {
+        FaultReporter.getInstance().addFault(
+          SUBSYSTEM_NAME,
+          "[System Check] Kicker Too Slow"
+        );
+      }
+    }
+  }
+
+  private Command getStatesCheckCommand() {
+    IntakeState[] desiredStateSequence = {
+      IntakeState.EMPTY,
+      IntakeState.NOTE_IN_INTAKE,
+      IntakeState.IN_BETWEEN_INTAKE_AND_DRUM,
+      IntakeState.NOTE_IN_DRUM,
+      IntakeState.IN_BETWEEN_DRUM_AND_KICKER,
+      IntakeState.NOTE_IN_KICKER
+    };
+    
+    ArrayList<IntakeState> actualStateSeqeunce = new ArrayList<>();
```

Comment:
This needs to be done as part of the command or else the list will have old value in it if the system test is run multiple times.

Reply:
I am not sure if this is what you meant, but the actualstatesequence arraylist is made as a brand new arraylist each time, so it would not have old values in it? This might mean something else though.